### PR TITLE
txverifier: fix bugs and improve error handling

### PR DESCRIPTION
**Pull request by johnsaigle**
- Sometimes From _can_ be set to the zero address when a token is minted. This restriction has been removed
- Made error handling around irrelevant receipts less noisy. Now there will not be any WARN logs when a receipt contains a LogMessagePublished event that was sent by someone other than the token bridge.
- Changed functions that process receipts to always return nil rather than the empty struct
- Added more unit tests to ensure that functions don't panic in nil input
- Added Validation function for TransferReceipts
- Add README for running the Transfer Verifier CLI tool
- Fix typo in variable name

### CODE REVIEW

File: node/pkg/txverifier/evmtypes.go [hunk 1]
```diff
@@ -518,7 +522,27 @@ type TransferReceipt struct {
...
+		return errors.Join(err, errors.New("parsed receipt's Deposits field is nil"))
+	}
+	if r.Transfers == nil {
+		return errors.Join(err, errors.New("parsed receipt's Transfers field is nil"))
+	}
+	if r.MessagePublications == nil {
+		return errors.Join(err, errors.New("parsed receipt's MessagePublications field is nil"))
+	}
+	if len(*r.MessagePublications) == 0 {
+		return errors.Join(err, errors.New("parsed receipt' has no Message Publications"))
```
Comment (pleasew8t) [inline node/pkg/txverifier/evmtypes.go:542]:
```suggestion
		return errors.Join(err, errors.New("parsed receipt has no Message Publications"))
```

File: node/pkg/txverifier/evm_test.go [hunk 2]
```diff
@@ -687,6 +687,39 @@ func TestProcessReceipt(t *testing.T) {
...
 
+func TestTransferReceiptValidate(t *testing.T) {
+	transferReceipt := &TransferReceipt{
+		Deposits:            &[]*NativeDeposit{},
+		Transfers:           &[]*ERC20Transfer{},
+		MessagePublications: &[]*LogMessagePublished{},
+	}
+	err := transferReceipt.Validate()
+	require.Error(t, err, "Validate must return error when its fields are nil")
+
```
Comment (pleasew8t) [inline node/pkg/txverifier/evm_test.go:698]: I don't think you are actually testing the function for `nil` input. This probably returns an error (expected) because of the length of `MessagePublications`, because the fields aren't `nil`. I'd recommend a testcase for each error message.

**Review by pleasew8t — DISMISSED** [review]

File: node/pkg/txverifier/evm.go [hunk 3]
```diff
@@ -86,10 +86,18 @@ func (tv *TransferVerifier[ethClient, Connector]) ProcessEvent(
 
 	// Parse raw transaction receipt into high-level struct containing transfer details.
 	transferReceipt, parseErr := tv.ParseReceipt(receipt)
-	if parseErr != nil || transferReceipt == nil {
-		tv.logger.Warn("error when parsing receipt. skipping validation",
-			zap.String("receipt hash", receipt.TxHash.String()),
-			zap.Error(parseErr))
+	if transferReceipt == nil {
```
Comment (bruce-riley) [inline node/pkg/txverifier/evm.go:89]: This change assumes that `tv.ParseReceipt` will never return an error when `transferReceipt` is not nil. Is that a safe assumption (maybe now, but what about in the future)?

Comment (johnsaigle) [reply node/pkg/txverifier/evm.go:89]: That's a good point, I'll add error-handling at this level at well.

**Review by bruce-riley — DISMISSED** [review]

File: node/pkg/txverifier/evm.go (hunk 3 above)
Comment (johnsaigle) [reply node/pkg/txverifier/evm.go:89]: Added in the most recent commit.

**Review by bruce-riley — APPROVED** [review]

**Review by evan-gray — APPROVED** [review]
only reviewed `cspell-custom-words.txt` changes
